shim/src/unsafeRec.js

```diff
@@ -10,8 +10,8 @@ import { freeze } from './commons';
 // this feature will be provided by the underlying engine instead.
 
 // Platform detection.
-const isNode = typeof exports === 'object' && typeof module !== 'undefined';
-const isBrowser = typeof document === 'object';
+const isBrowser = new Function('try {return this===window}catch(e){ return false}')(); // eslint-disable-line no-new-func
+const isNode = new Function('try {return this===global}catch(e){ return false}')(); // eslint-disable-line no-new-func
```

Comment:
and why not `typeof process !== 'undefined'`?

Reply:
Webpack tries to shim `process` if it sees this check, so this fix gives the original error of "unexpected platform, unable to create Realm"

Reply:
Gotcha, that does make sense.